### PR TITLE
[DROOLS-7538] IdentityHashMap inverseStorage

### PR DESCRIPTION
**Pull request by nprentza**
[DROOLS-7538](https://issues.redhat.com/browse/DROOLS-7538)

### CODE REVIEW

**Comment by kie-ci4** [comment]
**(tests) - kogito-quarkus-examples job** `#919` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT= 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/drools/main/.ci/buildchain-config.yaml' -o 'bc' -p kiegroup/kogito-examples -u https://github.com/kiegroup/drools/pull/5501 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/drools/job/main/job/pullrequest/job/drools.tests.downstream.kogito-quarkus-examples/919/display/redirect

**Test results:**
- PASSED: 166
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/drools/job/main/job/pullrequest/job/drools.tests.downstream.kogito-quarkus-examples/919/testReport/org.kie.kogito.quarkus.outbox/OutboxIT/___/">org.kie.kogito.quarkus.outbox.OutboxIT.(?)</a></summary>
org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 33306] should be listening)
</details>

File: drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableRefObjectStore.java [hunk 1]
```diff
@@ -97,12 +111,13 @@ private StoredObject setReferencedObjects(StoredObject object) {
     }
 
     private Long fromObjectToFactHandleId(Object object) {
-        for (Long key : this.storage.keySet()) {
+        return this.inverseStorage.get(object);
+        /*for (Long key : this.storage.keySet()) {
```
Comment (mariofusco) [inline drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableRefObjectStore.java]: Please remove commented out code.

File: drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableRefObjectStore.java [hunk 2]
```diff
@@ -31,18 +32,29 @@
...
     private Map<String, Long> uniqueObjectTypesInStore;  // object type name, occurances
+    private IdentityHashMap<Object, Long> inverseStorage;
 
     public SimpleSerializationReliableRefObjectStore(Storage<Long, StoredObject> storage) {
         super(storage);
         uniqueObjectTypesInStore = new HashMap<>();
-        if (storage.isEmpty()) {
-            this.storage = storage;
+        setInverseStorage(storage);
+        /*if (storage.isEmpty()) {
```
Comment (mariofusco) [inline drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableRefObjectStore.java]: Please remove commented out code.

**Review by mariofusco — APPROVED** [review]

File: drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableRefObjectStore.java (hunk 2 above)
Comment (nprentza) [reply drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableRefObjectStore.java]: Commented out code removed.

File: drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableRefObjectStore.java (hunk 1 above)
Comment (nprentza) [reply drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/SimpleSerializationReliableRefObjectStore.java]: Commented out code removed.

**Review by tkobayas — APPROVED** [review]

**Pull request merged**